aptos/contracts/sources/wormhole.move

```diff
@@ -1,55 +1,108 @@
 module wormhole::wormhole {
     use aptos_framework::account;
     use wormhole::structs::{create_guardian, create_guardian_set};
-    use wormhole::state::{
-        init_message_handles,
-        init_wormhole_state,
-        store_guardian_set,
-        set_governance_contract,
-        set_chain_id,
-        set_governance_chain_id,
-    };
+    use wormhole::state;
     use deployer::deployer;
     use wormhole::u16;
-    use wormhole::u32;
+    use wormhole::u32::{Self, U32};
+    use wormhole::emitter;
 
-    friend wormhole::contract_upgrade;
+// -----------------------------------------------------------------------------
+// Sending messages
 
-    // TODO(csongor): maybe merge the different capabilities into the same key
-    // to reduce storage access
+    public entry fun publish_message(
```

Comment:
maybe get rid of "entry" here, because this function can't be called from the client? (because of the non-primitive-type arg emitter)

Reply:
I think I read somewhere that the only args you can pass in to entry functions are primitives (bools, addresses, uints...)